src/lib/journey/callbacks/select-idp/select-idp.svelte

```diff
@@ -0,0 +1,66 @@
+<script lang="ts">
+  import type { SelectIdPCallback } from '@forgerock/javascript-sdk';
+  import AppleIcon from '../../../components/icons/apple-icon.svelte';
+  import Button from '$components/primitives/button/button.svelte';
+  import Grid from '$components/primitives/grid/grid.svelte';
+  import { interpolate, textToKey } from '$lib/_utilities/i18n.utilities';
+
+  import type {
+    CallbackMetadata,
+    SelfSubmitFunction,
+    StepMetadata,
+  } from '$journey/journey.interfaces';
+  import type { Style } from '$lib/style.store';
+  import type { Maybe } from '$lib/interfaces';
+
+  export let callback: SelectIdPCallback;
+  export let callbackMetadata: CallbackMetadata;
+  export let selfSubmitFunction: Maybe<SelfSubmitFunction> = null;
+  export let stepMetadata: StepMetadata;
+  export let style: Style = {};
+
+  let inputName: string;
+  let label: string;
+  let idps: { value: string; text: string }[];
+  let buttonStyle: 'outline' | 'primary' | 'secondary' | undefined;
+
+  /**
+   * @function setButtonValue - Sets the value on the callback on button click
+   * @param {number} index
+   */
+  function setBtnValue(value: string) {
+    callback.setProvider(value);
+    callbackMetadata.isReadyForSubmission = true;
+    selfSubmitFunction && selfSubmitFunction();
+  }
+
+  // Assume that clicking "next" will indicate the user wants to use local authentication
+  if (callbackMetadata.social?.isLocalAuthenticationEnabled) {
+    callback.setProvider('localAuthentication');
+  }
```

Comment:
As the above comment says, I think the cleanest / simplest way to show a "Local Authentication" option to the user is to by default set the provider to be "Local Authentication", and let the "Next" button on the page handle it. It means that we don't need to worry about there being another "submit" button on the page and that button displaying below all other callbacks, since "Next" already does this. Also, this is what Platform UI provides.

Reply:
I think this makes sense. I'll continue thinking on this, but I like how this keeps callbacks independent.